core-web/libs/edit-content/src/lib/fields/dot-edit-content-text-field/dot-edit-content-text-field.component.spec.ts

```diff
@@ -0,0 +1,111 @@
+import { test, describe } from '@jest/globals';
+import { Spectator, byTestId, createComponentFactory } from '@ngneat/spectator';
+
+import { CommonModule } from '@angular/common';
+import { ReactiveFormsModule, ControlContainer, FormGroupDirective } from '@angular/forms';
+
+import { InputTextModule } from 'primeng/inputtext';
+
+import { DotFieldRequiredDirective } from '@dotcms/ui';
+
+import { DotEditContentTextFieldComponent } from './dot-edit-content-text-field.component';
+
+import { FIELD_MOCK, createFormGroupDirectiveMock } from '../../utils/mocks';
+import { INPUT_TYPE } from '../models';
+
+describe('DotEditContentTextFieldComponent', () => {
+    let spectator: Spectator<DotEditContentTextFieldComponent>;
+    let textInput: Element;
+
+    const createComponent = createComponentFactory({
+        component: DotEditContentTextFieldComponent,
+        imports: [CommonModule, ReactiveFormsModule, InputTextModule, DotFieldRequiredDirective],
+        componentViewProviders: [
+            {
+                provide: ControlContainer,
+                useValue: createFormGroupDirectiveMock()
+            }
+        ],
+        providers: [FormGroupDirective]
+    });
+    beforeEach(() => {
+        spectator = createComponent({
+            props: {
+                field: FIELD_MOCK
+            }
+        });
+
+        textInput = spectator.query(byTestId('input-' + FIELD_MOCK.variable));
+    });
+
+    it('should render', () => {
+        expect(textInput).toBeTruthy();
+    });
+
+    test.each([
+        {
+            variable: FIELD_MOCK.variable,
+            attribute: 'id'
+        },
+        {
+            variable: FIELD_MOCK.variable,
+            attribute: 'ng-reflect-name'
+        }
+    ])('should have the $variable as $attribute', ({ variable, attribute }) => {
+        expect(textInput.getAttribute(attribute)).toBe(variable);
+    });
+
+    describe.each([
+        {
+            dataType: INPUT_TYPE.TEXT,
+            options: {
+                type: 'text',
+                inputMode: 'text'
+            }
+        },
+        {
+            dataType: INPUT_TYPE.INTEGER,
+            options: {
+                type: 'number',
+                inputMode: 'numeric',
+                step: 1
+            }
+        },
+        {
+            dataType: INPUT_TYPE.FLOAT,
+            options: {
+                type: 'number',
+                inputMode: 'decimal',
+                step: 0.1
+            }
+        }
```

Comment:
Dupe in the component, should we move this to a utils inside this same component and reuse?